src/core/server/internal_setup.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { InternalElasticsearchServiceSetup } from './elasticsearch';
+import { InternalHttpServiceSetup } from './http';
+import { InternalUiSettingsServiceSetup } from './ui_settings';
+import { ContextSetup } from './context';
+import { SavedObjectsServiceStart } from './saved_objects';
+
+/** @internal */
+export interface InternalCoreSetup {
+  context: ContextSetup;
+  http: InternalHttpServiceSetup;
+  elasticsearch: InternalElasticsearchServiceSetup;
+  uiSettings: InternalUiSettingsServiceSetup;
+}
+
```

Comment:
The `server/types.ts` files states `This module is intended for consumption by public to avoid import issues with server-side code`, so I created another file for our internal types. Tell me if this should be moved elsewhere.

Reply:
Separate file is fine to me, but I think public and server should be consistent. Also I think the name should probably be something like `internal_types` rather than `internal_setup`.